src/views/lighttable.c

```diff
@@ -2062,62 +2118,90 @@ static void _culling_prefetch(dt_view_t *self)
+        "SELECT m.imgid, b.aspect_ratio FROM memory.collected_images AS m, images AS b WHERE m.rowid < (SELECT "
+        "rowid FROM "
+        "memory.collected_images WHERE imgid = %d) AND m.imgid = b.id ORDER BY m.rowid DESC LIMIT 1",
+        lib->slots[0].imgid);
+    sqlite3_stmt *stmt;
+    DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), query, -1, &stmt, NULL);
+
+    if(stmt != NULL)
     {
-      imgid_prev = sqlite3_column_int(stmt, 0);
+      if(sqlite3_step(stmt) == SQLITE_ROW)
+      {
+        lib->culling_previous.imgid = sqlite3_column_int(stmt, 0);
+        double aspect_ratio = sqlite3_column_double(stmt, 1);
+        if(!aspect_ratio || aspect_ratio < 0.0001)
+        {
+          aspect_ratio = dt_image_set_aspect_ratio(lib->culling_previous.imgid);
+          // if an error occurs, let's use 1:1 value
+          if(aspect_ratio < 0.0001) aspect_ratio = 1.0;
+        }
+        lib->culling_previous.aspect_ratio = aspect_ratio;
+      }
+      sqlite3_finalize(stmt);
     }
-    sqlite3_finalize(stmt);
-  }
-  g_free(query);
+    g_free(query);
 
-  // we get the next image id
-  int imgid_next = -1;
-  query = dt_util_dstrcat(NULL,
-                          "SELECT imgid FROM memory.collected_images WHERE rowid > (SELECT rowid FROM "
-                          "memory.collected_images WHERE imgid = %d) ORDER BY rowid LIMIT 1",
-                          lib->slots[lib->slots_count - 1].imgid);
-  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), query, -1, &stmt, NULL);
-  if(stmt != NULL)
-  {
-    if(sqlite3_step(stmt) == SQLITE_ROW)
+    // and we prefetch the image
+    if(lib->culling_previous.imgid >= 0)
     {
-      imgid_next = sqlite3_column_int(stmt, 0);
+      dt_mipmap_size_t mip = dt_mipmap_cache_get_matching_size(
+          darktable.mipmap_cache, imgwd * lib->slots[0].width * fz, imgwd * lib->slots[0].height * fz);
+
+      if(mip < DT_MIPMAP_8)
+        dt_mipmap_cache_get(darktable.mipmap_cache, NULL, lib->culling_previous.imgid, mip, DT_MIPMAP_PREFETCH,
+                            'r');
     }
-    sqlite3_finalize(stmt);
   }
-  g_free(query);
 
-  // and we pretetch the mipmaps
-  const float imgwd = 0.97;
-  const float fz = (lib->full_zoom > 1.0f) ? lib->full_zoom : 1.0f;
-
-  if(imgid_next >= 0)
+  // we get the next image infos
+  if(lib->culling_next.imgid == -1)
   {
-    dt_mipmap_size_t mip = dt_mipmap_cache_get_matching_size(darktable.mipmap_cache,
-                                                             imgwd * lib->slots[lib->slots_count - 1].width * fz,
-                                                             imgwd * lib->slots[lib->slots_count - 1].height * fz);
-
-    if(mip < DT_MIPMAP_8)
-      dt_mipmap_cache_get(darktable.mipmap_cache, NULL, imgid_next, mip, DT_MIPMAP_PREFETCH, 'r');
-  }
+    gchar *query
+        = dt_util_dstrcat(NULL,
+                          "SELECT m.imgid, b.aspect_ratio FROM memory.collected_images AS m, images AS b WHERE "
```

Comment:
I understand that this was the case in this code, but I think we should try to have better readability for the SQL commands. I'm doing:

```
   "SELECT ..."
   " FROM ..."
   " WHERE ..."
   " ORDER_BY .."
```

Of course if the FROM is small it can be on the same line as SELECT, but having the others starting on a new line makes things lot more readable.

Reply:
done (in other queries too)